Formula/hblock.rb

```diff
@@ -0,0 +1,17 @@
+class Hblock < Formula
+  desc "Adblocker that creates a hosts file from multiple sources"
+  homepage "https://github.com/hectorm/hblock"
+  url "https://github.com/hectorm/hblock/archive/v3.1.3.tar.gz"
+  sha256 "e7dad9febc5205f19b36ce9268a41374931d27e0195a2d7504b62c2d1721ab7f"
+  license "MIT"
+
+  uses_from_macos "curl"
+
+  def install
+    system "make", "install", "prefix=#{prefix}", "bindir=#{bin}", "mandir=#{man}"
+  end
+
+  test do
+    system "#{bin}/hblock", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Would this be enough?